readme.md

```diff
@@ -0,0 +1,177 @@
+This document will help you to understand how to make use of CargoDash. CargoDash is a diagnostics tool for developing autonomous vehicles and is written in Python code (version 3.8.5). For any questions please feel free to create an issue.
+
+## Python Dependencies
+```
+    aiohttp v3.7.3
+    can v3.3.4
+    canopen v1.1.0
+    numpy v1.19.4
+    pyyaml v5.3.1
+```
+## Installation
+1. Clone the repository
+```
+    git clone https://github.com/ZeroPoint095/CargoDash
+```
+2. Install the Python dependencies with pip3
+```
+    pip3 install -r requirements.txt
+```
+3. Run startup.sh to use CargoDash
+```
+    ./startup.sh
+```  
+
+## Introduction
+CargoDash is a tool that listens to incoming messages and sets the values of these messages into user-defined nodes. These user-defined nodes can be requested with CargoDash's HTTP-server. Besides this, it logs all raw incoming messages into a buffered logger. 
+
+For now, CargoDash works only with CanOpen messages but with the architecture of CargoDash, you should be able to add your desired message protocol.
+
+The buffered logger can be requested to log all messages from the buffer on command. The buffer is circular and records the messages continuously. When the buffer is full then it continues by overwriting the oldest messages.  For example, when the buffer just filled in index 1024 and it has a shape from index 0 to index 1024 then it starts over again from index 0. The buffer gets filled from 0 to 1024 again. The logger has many useful purposes. You can request the buffer on a dangerous event or just when a user wants to see the log.
+
+## Our usage
+We created CargoDash to listen to a CanOpen network and use it for the Smarterdam project. The Smarterdam project is funded by Rotterdam University of Applied Sciences. CargoDash is able to track all incoming CanOpen messages and we are able to give more meaning to messages that have been send and received.
+
+In our application, we created a network of Arduino's that are communicating with a Raspberry Pi that is running the CargoDash code. For the communication between the Arduino's and the Raspberry Pi, we created a can bus network. All these micro-controllers are communicating with the CanOpen protocol. For the Arduino's to communicate in CanOpen, we forked [jgeisler0303's](https://github.com/jgeisler0303/CANFestivino) repository. We fixed some bugs and adapted it more to our use case. Our repository can be found [here](https://github.com/ZeroPoint095/CANFestivino).
+
+## Using the configuration file for CanOpen (config.yaml)
+
+Inside the code block below you can read a detailed description of which configurations that CargoDash uses. In general, we prefer that CargoDash can be configured from one file such as the config.yaml file instead of multiple configuration places. This makes CargoDash more user-friendly and centralized. 
+```yaml
+    # canopen_vcan is a configuration and is a configuration example for the CanOpen protocol.
```

Comment:
`is a configuration and is a configuration` dubbel op

Reply:
jup je hebt gelijk